Automattic/wp-calypso
# Domains: Use new domain transfer flow when starting a pending transfer

#### Changes proposed in this Pull Request
Replaces the previous precheck call with the new transfer flow.
Some changes were made to the container components in order to accept the `transfer` state as the initial one.

#### Preview
##### Normal state
![image](https://user-images.githubusercontent.com/18705930/136492415-41b704dd-4441-44a9-871c-237ec8a66960.png)

##### Domain not transferable (a sanity check, in case URL is manually changed by user)
![image](https://user-images.githubusercontent.com/18705930/136492630-a8649702-bcf9-4780-b791-405a31ec66cc.png)

These are the only changes specific to this flow: all the other parts are the same.

#### Testing instructions
- With a domain in "pending transfer" state, check that clicking on the "Start transfer" button no longer shows the precheck - and that the transfer within the new flow is successfully started. 
- Check that this flow is not able to proceed if a domain that can't be transferred is provided: 
  - Change the query param to a domain that is not transferrable;
  - Confirm that the message displayed is the same that the one in the "Preview" section of this PR's description.
- Check that forcing a transfer to a domain different than the one previously bought is not possible:
  - Change the query param to a different (and transferrable) domain than the one previously bought;
  - Confirm that the final step of the flow returns an error - even if you provide the correct auth code.
